libs/books/feature/src/lib/reading-list/reading-list.component.ts

```diff
@@ -10,9 +12,18 @@ import { getReadingList, removeFromReadingList } from '@tmo/books/data-access';
 export class ReadingListComponent {
   readingList$ = this.store.select(getReadingList);
 
-  constructor(private readonly store: Store) {}
-
+  constructor(private readonly store: Store,
+    private snackBar: MatSnackBar) {}
+  
+    openSnackbar(message:string, action:string, item: Book) {
+      let snackBarRef = this.snackBar.open(message, action,{duration: 6000});
+    
+      snackBarRef.onAction().subscribe(() => {
+          this.store.dispatch(addToReadingList({book:item}));
+      })
+    }  
   removeFromReadingList(item) {
     this.store.dispatch(removeFromReadingList({ item }));
+    this.openSnackbar('Removed from Reading List', 'Undo',item);
```

Comment:
Same here. You are opening the snackbar the moment the user removes a book from the reading list. How do you know that the book was successfully removed, to allow it to be added again (undo)? You should make sure that the `removeFromReadingList` was successful, in order to allow its undoing, or else you could run into errors! You could try a different place to open the snackbar from, where all these actions are available! (hint: effects)